core/src/utils/racy_lock.rs

```diff
@@ -0,0 +1,147 @@
+use alloc::boxed::Box;
+use core::{
+    ops::Deref,
+    ptr,
+    sync::atomic::{AtomicPtr, Ordering},
+};
+
+/// Thread-safe, non-blocking, "first one wins" flavor of `once_cell::sync::OnceCell`
+/// with the same interface as `std::sync::LazyLock`.
+///
+/// The underlying implementation is based on `once_cell::sync::race::OnceBox` which relies on
```

Comment:
I would probably include a link here (I think https://github.com/matklad/once_cell/blob/c48d3c2c01de926228aea2ac1d03672b4ce160c1/src/race.rs#L294 is the right one?).

Reply:
I would also add an `# Acknowledgements` section to `core/README.md` in which we explicitly state that `core/src/utils/racy_lock.rs` has some code taken from `once_cell`.